recipes/gcc/all/conanfile.py

```diff
@@ -1,126 +1,162 @@
-            "--with-system-zlib",
-            "--with-gmp=%s" % self.deps_cpp_info['gmp'].rootpath,
-            '--with-mpc=%s' % self.deps_cpp_info["mpc"].rootpath,
-            "--with-mpfr=%s" % self.deps_cpp_info["mpfr"].rootpath,
-            "--without-isl",
-            "--libdir=%s" % libdir,
-            '--with-pkgversion=%s' % pkgversion,
-            "--program-suffix=-%s" % self.version,
-            "--with-bugurl=%s" % bugurl
-        ]
-        if self.settings.os == "Macos":
-            xcrun = tools.XCRun(self.settings)
-            args.extend([
-                '--with-native-system-header-dir=/usr/include',
-                "--with-sysroot={}".format(xcrun.sdk_path)
-            ])
-        self._autotools.libs = []  # otherwise causes config.log to fail finding -lmpc
+    def configure(self):
         if self.settings.compiler in ["clang", "apple-clang"]:
-            # xgcc: error: unrecognized command-line option -stdlib=libc++
-            if self.settings.compiler.libcxx == "libc++":
-                self._autotools.cxx_flags.remove("-stdlib=libc++")
-            elif self.settings.compiler.libcxx in ["libstdc++", "libstdc++11"]:
-                self._autotools.cxx_flags.remove("-stdlib=libstdc++")
-        self._autotools.configure(args=args, configure_dir=self._source_subfolder)
-        return self._autotools
-
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+            # Can't remove this from cxxflags with autotools - so get rid of it
+            del self.settings.compiler.libcxx
+
+    def build_requirements(self):
+        self.tool_requires("binutils/2.38")
+        self.tool_requires("flex/2.6.4")
 
     def requirements(self):
         self.requires("mpc/1.2.0")
         self.requires("mpfr/4.1.0")
-        self.requires("gmp/6.2.0")
-        self.requires("zlib/1.2.11")
+        self.requires("gmp/6.2.1")
+        self.requires("zlib/1.2.12")
+        self.requires("isl/0.24")
 
-    def configure(self):
-        if self.settings.os == "Windows":
-            raise ConanInvalidConfiguration("Windows builds aren't supported (yet), sorry")
-        if tools.cross_building(self.settings):
+    def package_id(self):
+        del self.info.settings.compiler
+
+    def validate(self):
+        if self.info.settings.os == "Windows":
+            raise ConanInvalidConfiguration(
+                "Windows builds aren't supported (yet), sorry"
+            )
+        if cross_building(self):
             raise ConanInvalidConfiguration("no cross-building support (yet), sorry")
 
-    def source(self):
-        tools.get(**self.conan_data["sources"][self.version])
-        extracted_dir = "gcc-%s" % self.version
-        os.rename(extracted_dir, self._source_subfolder)
+    def layout(self):
+        basic_layout(self, src_folder="source")
+
+    def generate(self):
+        # Ensure binutils and flex are on the path
+        buildenv = VirtualBuildEnv(self)
```

Comment:
This is automatic in 2.0 fyi and there a conf for 1.x to match the behavior :)

Reply:
Yep, appreciated. I think my preference would be to force it here rather than relying on the conf value, and remove it when conan 2.0 rolls around. Is that acceptable?